mdagent/tools/maketools.py

```diff
@@ -186,15 +185,21 @@ class CreateNewToolInputSchema(BaseModel):
         description="""List of all tools you have access to. Such as
         this tool, 'ExecuteSkill', 'SkillRetrieval', and maybe `Name2PDBTool`, etc."""
     )
+    execute: Optional[bool] = Field(
+        description="Whether to execute the new tool or not. default: True"
+    )
```

Comment:
you can include the default value like: 
```{python}
Field(True, description="Whether to execute the new tool or not. default: True")
```
That way, you don't need to specify the value later on when calling _run

Reply:
updated in latest commits